### PR TITLE
skeptipay models and specs

### PR DESCRIPTION
**Pull request by KwaneleNdenze**

### CODE REVIEW

File: db/migrate/20220622092155_create_payments.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreatePayments < ActiveRecord::Migration[5.2]
+  def change
+    create_table :payments do |t|
+      t.string :amount
+      t.text :description
+      t.string :verification_code
+      t.string :status
```
Comment (thatosmk) [inline db/migrate/20220622092155_create_payments.rb:7]: make this an enum (which will be an `integer`). Have a look and see how they do enums in ruby

File: db/migrate/20220622093223_create_withdrawals.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class CreateWithdrawals < ActiveRecord::Migration[5.2]
+  def change
+    create_table :withdrawals do |t|
+      t.string :amount
+      t.string :status
```
Comment (thatosmk) [inline db/migrate/20220622093223_create_withdrawals.rb:5]: make this an enum as well

File: db/migrate/20220622093329_create_notifications.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class CreateNotifications < ActiveRecord::Migration[5.2]
+  def change
+    create_table :notifications do |t|
+      t.string :recipient
+      t.text :text
```
Comment (thatosmk) [inline db/migrate/20220622093329_create_notifications.rb:5]: call it `content` or `body` or something more specific

**Comment by thatosmk** [comment]
looks great, but also, have a look at having an `account` model that will have a `user_id` and a user will have an account and that account will have invoices, withdrawals etc

**Review by thatosmk — CHANGES_REQUESTED** [review]
add enums and perhaps think about having an accounts table.

File: db/migrate/20220622093223_create_withdrawals.rb (hunk 2 above)
Comment (KwaneleNdenze) [reply db/migrate/20220622093223_create_withdrawals.rb:5]: Made this change

File: db/migrate/20220622093329_create_notifications.rb (hunk 3 above)
Comment (KwaneleNdenze) [reply db/migrate/20220622093329_create_notifications.rb:5]: Made this change.

**Comment by KwaneleNdenze** [comment]
> 

Made the change, please check if my implementation of the associations between user and account is correct

File: app/models/notification.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class Notification < ApplicationRecord
+  belongs_to :account
+
+  enum status: [:unread, :read]
```
[the remark this answers is not in the document]
Comment (thatosmk) [reply app/models/notification.rb:4]: also fix indentation in this file, it is not uniform

File: app/models/payment.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+class Payment < ApplicationRecord
+  belongs_to :account
+  
+  enum status: [:requested, :approved]
```
Comment (thatosmk) [inline app/models/payment.rb:4]: same comment as above regarding enums

File: app/models/payment.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+  after_initialize :set_default_status, :if => :new_record?
+
+    def set_default_status
+      self.status ||= :requested
+    end
```
Comment (thatosmk) [inline app/models/payment.rb:10]: you can add `default: 0` to the migration when you add the status column.

File: app/models/payment.rb (hunk 5 above)
Comment (thatosmk) [reply app/models/payment.rb:4]: so no need to do this

File: app/models/user.rb [hunk 7]
```diff
@@ -1,4 +1,5 @@
 class User < ApplicationRecord   
+   has_one :account
```
Comment (thatosmk) [inline app/models/user.rb:2]: correct, so fix the association in the `account` model

File: app/models/payment.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
+class Payment < ApplicationRecord
+  belongs_to :account
+  
+  enum status: {requested: 0, approved: 0}
```
Comment (thatosmk) [inline app/models/payment.rb:4]: cannot have the same integer value

File: db/migrate/20220622092155_create_payments.rb (hunk 1 above)
Comment (thatosmk) [inline db/migrate/20220622092155_create_payments.rb:7]: make this an integer with default 0, then remove all your set default status methods

File: app/models/notification.rb [hunk 9]
```diff
@@ -0,0 +1,14 @@
+class Notification < ApplicationRecord
+  belongs_to :account
+
+  enum status: {unread: 0, read: 1}
+
+  after_initialize :set_default_status, :if => :new_record?
+
+  def set_default_status
+    self.status ||= :unread
+  end
```
Comment (thatosmk) [inline app/models/notification.rb:10]:
```suggestion
```

File: app/models/payment.rb [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+  after_initialize :set_default_status, :if => :new_record?
+
+  def set_default_status
+    self.status ||= :requested
+  end
```
Comment (thatosmk) [inline app/models/payment.rb:10]:
```suggestion
```

**Review by thatosmk — APPROVED** [review]
Great work dude

**Pull request merged**